tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts

```diff
@@ -0,0 +1,178 @@
+    int tileColA = int(gl_LocalInvocationID.x);
+    int tileRowB = int(gl_LocalInvocationID.y);
+    for (int t = 0; t < numTiles; t++) {
+      if (t == 0) {
+        if (tileRow < TileAOuter) {
+          // Load one tile of A and B into local memory.
+          mm_Asub1[tileRow][tileColA] =
+              mm_readA((globalRow - tileRow) / 2 + tileRow, globalColA);
+          globalColA += TileInner;
+          mm_Bsub1[tileRowB][tileCol] =
+              mm_readB(t * TileInner + tileRowB, globalCol);
+        }
+      } else {
+        if (tileRow < TileAOuter) {
+          // Load one tile of A and B into local memory.
+          mm_Asub1[tileRow][tileColA] =
+              mm_readA((globalRow - tileRow) / 2 + tileRow, globalColA);
+          globalColA += TileInner;
+          mm_Bsub1[tileRowB][tileCol] =
+              mm_readB(t * TileInner + tileRowB, globalCol);
+        } else {
+          // Compute acc values for a single thread.
+          for (int k = 0; k < TileInner; k++) {
+            acc += mm_Asub2[tileRow - TileAOuter][k] * mm_Bsub2[k][tileCol];
+          }
+        }
+      }
+      barrier();
+      if (t != 0) {
+        t++;
+      }
+
+      if (t < numTiles) {
+        if (tileRow < TileAOuter) {
+          // Load one tile of A and B into local memory.
+          mm_Asub2[tileRow][tileColA] =
+              mm_readA((globalRow - tileRow) / 2 + tileRow, globalColA);
+          mm_Bsub2[tileRowB][tileCol] =
+              mm_readB(t * TileInner + tileRowB, globalCol);
+        } else {
+          // Compute acc values for a single thread.
+          for (int k = 0; k < TileInner; k++) {
+            acc += mm_Asub1[tileRow - TileAOuter][k] * mm_Bsub1[k][tileCol];
+          }
+        }
+      }
+      barrier();
+    }
+    if (tileRow >= TileAOuter) {
+      mm_write((globalRow - tileRow) / 2 + tileRow - TileAOuter,
+          globalCol, acc);
+    }
+  }
+  `;
+}
+
+export class MatMulSmallOutputSizeProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[], y: number[], z: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['A', 'B'];
+  workGroupSize: [number, number, number] = [8, 16, 1];
+  addBias: boolean;
+  activation: backend_util.Activation;
+  hasPreluActivationWeights: boolean;
+
+  constructor(
+      aShape: [number, number, number], bShape: [number, number, number],
+      outputShape: [number, number, number], bias: TensorInfo = null,
+      activation: backend_util.Activation = null,
+      preluActivationWeights: TensorInfo = null) {
+    util.assert(aShape[1] <= 16 || bShape[2] <= 16,
+      () => 'This program can be only used when A width is small.');
+    this.outputShape = outputShape;
+
+    this.dispatchLayout = {x: [2], y: [1], z: [0]};
+    this.dispatch = [Math.ceil(outputShape[2] / this.workGroupSize[0]),
+        Math.ceil(outputShape[1] * 2 / this.workGroupSize[1]), outputShape[0]];
```

Comment:
Please add a comment above the line to explain why you need `outputShape[1] * 2` here. Also add a comment above `mm_matMul` to briefly describe your algorithm's advantages.